rust/core-lib/src/tabs.rs

```diff
@@ -542,11 +544,58 @@ impl CoreState {
     fn finalize_new_views(&mut self) {
         let to_start = mem::replace(&mut self.pending_views, Vec::new());
         to_start.iter().for_each(|(id, config)| {
+            let config = &self.detect_whitespace(*id, config);
             let mut edit_ctx = self.make_context(*id).unwrap();
             edit_ctx.finish_init(config);
         });
     }
 
+    // Detects indentation settings and returns an updated config if autodetect is on.
+    fn detect_whitespace(&mut self, id: ViewId, config: &Table) -> Table {
```

Comment:
I think we can definitely clean this up, but I'm going to let that be follow-up, want to just focus on the implementation for now. 👍